Desktop_Interface/genericusbdriver.cpp

```diff
@@ -87,13 +86,15 @@ void genericUsbDriver::setFunctionGen(functionGenControl::ChannelID channelID, f
       //////////////////////////////////////
 
     //For recalling on crash.
-    if (channelID == ChannelID::CH1)
-		fGenPtr_CH1 = fGenControl;
-    else
-		fGenPtr_CH2 = fGenControl;
+	fGenPtrData[(int)channelID] = fGenControlData;
+
+	useFunctionGenData(channelID);
+}
 
+void genericUsbDriver::useFunctionGenData(ChannelID channelID)
```

Comment:
I'm not sure "use" is the best word here.  Maybe "send"?

Reply:
Ah yes, i though i had left a comment about that, I wasn't really sure what to name it.